Update some of the C# documentation for 4.0 (#6681)
godotengine/godot-docs: tutorials/scripting/c_sharp/c_sharp_differences.rst

located under ``Mathf`` as ``Abs``, ``Acos``, ``Asin``, ``Atan`` and ``Atan2``.
The ``PI`` constant can be found as ``Mathf.Pi``.

> C# also provides the static `System.Math`_ and `System.MathF`_ classes that may

Reviewer: ```suggestion
C# also provides static `System.Math`_ and `System.MathF`_ classes that may
```

        }
    }

> `@export` annotation
> --------------------

Reviewer: ```suggestion
``@export`` annotation
----------------------
```

        private string _icon;
    }

See also: :ref:`doc_c_sharp_exports`.

> Signal keyword
> --------------

Reviewer: ```suggestion
``signal`` keyword
------------------
```

Should we do it like this for consistency with the annotation formatting?

Author: Makes sense to me.

Variant
-------

> ``Godot.Variant`` is used as the variant type, any variant-compatible type can be converted from/to it.
> Consider avoiding ``Godot.Variant`` and take advantage of C#'s type safety when possible,
> only use ``Godot.Variant`` when necessary to use Godot untyped APIs.

Reviewer: ```suggestion
``Godot.Variant`` is used to represent the Godot's native :ref:`Variant <doc_variant_class>` type. Any Variant-compatible type can be converted from/to it.
We recommend avoiding ``Godot.Variant`` unless it is necessary to interact with untyped engine APIs.
Take advantage of C#'s type safety when possible.
```

> The ``Variant.As{TYPE}`` methods or the generic ``Variant.As<T>`` method can be used to convert
> a ``Godot.Variant`` to a C# type. Since the ``Godot.Variant`` type contains implicit conversions
> defined for all the supported types calling these methods directly is usually not necessary.

Reviewer: ```suggestion
Any of ``Variant.As{TYPE}`` methods and a generic ``Variant.As<T>`` method can be used to convert
a ``Godot.Variant`` to a C# type. Since the ``Godot.Variant`` type contains implicit conversions
defined for all the supported types calling these methods directly is usually not necessary.
```

Author: This doesn't sound right to me, there is only one `Variant.As<T>` method so I feel like using `a` is wrong here.

Reviewer: For some reason it trips me here, but I guess this is because I intuitively read "generic" in its literal sense, not as a CS term :) Sure, you can keep the article for the generic method (and the same for the snippet below). (but the rest of the suggestion stands).

> Use the ``CreateFrom`` method overloads or the generic ``From<T>`` method to convert a C# type
> to a ``Godot.Variant``.

Reviewer: ```suggestion
Use ``CreateFrom`` method overloads or a generic ``From<T>`` method to convert a C# type
to a ``Godot.Variant``.
```

Communicating with other scripting languages
--------------------------------------------

This is explained extensively in :ref:`doc_cross_language_scripting`.

.. _doc_c_sharp_differences_await:

> Await keyword
> -------------

Reviewer: Same concern as with the signal.

```suggestion
``await`` keyword
-----------------
```

> Something similar to GDScript's ``await`` keyword can be achieved with C#'s
> `await keyword <https://docs.microsoft.com/en-US/dotnet/csharp/language-reference/keywords/await>`_.

Reviewer: ```suggestion
Something similar to the GDScript's ``await`` keyword can be achieved with the C#'s
`await keyword <https://docs.microsoft.com/en-US/dotnet/csharp/language-reference/keywords/await>`_.
```

Author: For some reason it sounds weird to me to use the `the` article with `'s`. But if you think it's more correct like this.

Reviewer: I guess if it's apostrophe'd, then we don't need an article. You can ignore this.

> The C# ``await`` keyword can be used with any awaitable expression, it's commonly

Reviewer: ```suggestion
The ``await`` keyword in C# can be used with any awaitable expression. It's commonly
```

used with operands of the types `Task`_, `Task<TResult>`_, `ValueTask`_, or `ValueTask<TResult>`_.

An expression ``t`` is awaitable if one of the following holds:

* ``t`` is of compile-time type ``dynamic``.
* ``t`` has an accessible instance or extension method called ``GetAwaiter`` with no
  parameters and no type parameters, and a return type ``A`` for which all of the
  following hold:

  * ``A`` implements the interface ``System.Runtime.CompilerServices.INotifyCompletion``.
  * ``A`` has an accessible, readable instance property ``IsCompleted`` of type ``bool``.
  * ``A`` has an accessible instance method ``GetResult`` with no parameters and no type
    parameters.

.. _Task: https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task
.. _Task<TResult>: https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task-1
.. _ValueTask: https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask
.. _ValueTask<TResult>: https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask-1

> The equivalent of await on signal can be achieved with the ``await`` keyword and

Reviewer: ```suggestion
An equivalent of awaiting a signal in GDScript can be achieved with the ``await`` keyword and
```
